### ISSUE TITLE
Intersected buildings are not detected

### ISSUE THREAD

**Issue by vmartyanov** [openstreetmap/iD#10137]
### URL

https://www.openstreetmap.org/changeset/147971041

### How to reproduce the issue?

I think the screenshot is self-explaining:

![image](https://github.com/openstreetmap/iD/assets/9350107/f0bda873-b693-4115-822e-d547366ca978)

### Screenshot(s) or anything else?

Seems intersection is detected if and only if two lines intersects. But in this case segments have common nodes and this cause no lines intersection. But buildings are definitely intersected!

Maproulette challenge detects it, iD doesn't

### Which deployed environments do you see the issue in?

Released version at openstreetmap.org/edit

### What version numbers does this issue effect?

2.27.3

### Which browsers are you seeing this problem on?

Firefox

**Comment by k-yle** [comment]
As a side note, I had assumed this behaviour was intentional when I created #10120. If this is indeed a bug, then #10120 will need reworking

**Comment by vmartyanov** [comment]
Hm, #10120 has different shape: two squares with a common edge

**Comment by k-yle** [comment]
yeah #10120 is a different arrangement, but my "fix" for that issue assumes that your screenshot is a valid situation.

Anyway, I'm following this thread now, so I can update #10120 depending on what happens here

**Comment by tyrasd** [comment]
I don't think this is intentional. What's most likely missing is a simple point-in-polygon test for the vertices of the building(s), as the validator seems to only check for line-line intersections of the building's edges.
